src/Core/Domain/Common/Contracts/BaseEntityWith.cs

```diff
@@ -0,0 +1,6 @@
+﻿namespace DN.WebApi.Domain.Common.Contracts;
+
+public class BaseEntityWith<T> : BaseEntity

```

Comment:
No need to call this BaseEntityWith<T>, it can be called BaseEntity<T>. And I wouldn't derive from BaseEntity, but rather make BaseEntity implement BaseEntity<Guid>. This way the default is Guid, and you don't have to change a lot of the other code.

Reply:
yeah that's a good point, but I had another feature in mind (supporting view entities) that also derive from `BaseEntity`